engine/resources/mount.go

```diff
@@ -0,0 +1,687 @@
+			if !ok {
+				return nil
+			}
+			if err := obj.init.Read(event); err != nil {
+				return err
+			}
+		}
+
+		// do all our event sending all together to avoid duplicate msgs
+		if send {
+			send = false
+			if err := obj.init.Event(); err != nil {
+				return err // exit if requested
+			}
+		}
+	}
+}
+
+// fstabCheckApply checks /etc/fstab for entries corresponding to the resource
+// definition, and adds or deletes the entry as needed.
+func (obj *MountRes) fstabCheckApply(apply bool) (checkOK bool, err error) {
+	exists, err := fstabEntryExists(fstabPath, obj.mount)
+	if err != nil {
+		return false, errwrap.Wrapf(err, "error checking if fstab entry exists")
+	}
+
+	// if everything is as it should be, we're done
+	if (exists && obj.State == "exists") || (!exists && obj.State == "absent") {
+		return true, nil
+	}
+
+	if !apply {
+		return false, nil
+	}
+	obj.init.Logf("fstabCheckApply(%t)", apply)
+
+	if obj.State == "exists" {
+		if err := fstabEntryAdd(fstabPath, obj.mount); err != nil {
+			return false, errwrap.Wrapf(err, "error adding fstab entry: %+v", obj.mount)
+		}
+		return false, nil
+	}
+	if err := fstabEntryRemove(fstabPath, obj.mount); err != nil {
+		return false, errwrap.Wrapf(err, "error removing fstab entry: %+v", obj.mount)
+	}
+	return false, nil
+}
+
+// mountCheckApply checks if the defined resource is mounted, and mounts or
+// unmounts it according to the defined state.
+func (obj *MountRes) mountCheckApply(apply bool) (bool, error) {
+	exists, err := mountExists(procPath, obj.mount)
+	if err != nil {
+		return false, errwrap.Wrapf(err, "error checking if mount exists")
+	}
+
+	// if everything is as it should be, we're done
+	if (exists && obj.State == "exists") || (!exists && obj.State == "absent") {
+		return true, nil
+	}
+
+	if !apply {
+		return false, nil
+	}
+	obj.init.Logf("mountCheckApply(%t)", apply)
+
+	if obj.State == "exists" {
+		// Reload mounts from /etc/fstab by performing a `daemon-reload` and
+		// restarting `local-fs.target` and `remote-fs.target` units.
+		if err := mountReload(); err != nil {
+			return false, errwrap.Wrapf(err, "error reloading /etc/fstab")
+		}
+		return false, nil // we're done
+	}
+	// unmount the device
+	if err := unix.Unmount(obj.Name(), 0); err != nil {
+		return false, errwrap.Wrapf(err, "error unmounting %s", obj.Name())
+	}
+	return false, nil
+}
+
+// CheckApply is run to check the state and, if apply is true, to apply the
+// necessary changes to reach the desired state. This is run before Watch and
+// again if Watch finds a change occurring to the state.
+func (obj *MountRes) CheckApply(apply bool) (checkOK bool, err error) {
+	checkOK = true
+
+	if c, err := obj.fstabCheckApply(apply); err != nil {
+		return false, err
+	} else if !c {
+		checkOK = false
+	}
+
+	if c, err := obj.mountCheckApply(apply); err != nil {
+		return false, err
+	} else if !c {
+		checkOK = false
+	}
+
+	return checkOK, nil
+}
+
+// MountUID is a unique resource identifier.
```

Comment:
Can you move these things to be in the same order as the other resources please?